src/main.rs

```diff
@@ -33,23 +30,26 @@ cfg_if! {
             let addr = conf.leptos_options.site_addr;
 
             // Generate the list of routes in your Leptos App
-            let routes = generate_route_list(|| view! { <App/> });
+            let routes = generate_route_list(App);
 
             HttpServer::new(move || {
                 let leptos_options = &conf.leptos_options;
                 let site_root = &leptos_options.site_root;
                 let routes = &routes;
 
                 App::new()
-                    .route("/api/{tail:.*}", leptos_actix::handle_server_fns())
+                    .app_data(web::Data::new(Client::new()))
+                    .route("/api/{tail:.*}", web::route()
```

Comment:
Ask in the leptos dev chats if you haven't already about this. It doesn't seem right that this very simple case (posting a form and updating data based on its result), needs wrapping to our back-end, and server routing.

Reply:
![Screenshot_2023-10-04-15-19-18-99_572064f74bd5f9fa804b05334aa4f912](https://github.com/LemmyNet/lemmy-ui-leptos/assets/28871516/97a032c2-21df-47ad-98ec-597d21ba7a77)

I asked in discord and actix-proxy was suggested. If a request is being made, the host is the same, and no port is specified, the leptos server will catch the request and return a 404 if the request doesn't get routed to the leptos server. If not here, this will have to be handled somewhere.